docs/proposals/diagnose-experience.md

```diff
@@ -0,0 +1,247 @@
+
+#### Curl
+
+Curl only has a `--verbose` option to output verbose logs. No `--debug` option.
+
+#### Docker
+
+Docker has `--debug` and `--log-level` options to control debug logs output within different log levels, such as INFO, DEBUG, WARN, etc. No `--verbose` option. Docker has its own daemon service running in local so its logs might be much more complex.
+
+#### Kubectl
+
+Kubectl has a command `kubectl logs` to show logs of resource objects such as Pod and container. No separate flags for verbose output and debug logs.
+
+## Examples in ORAS
+
+This section lists the current behaviors of ORAS verbose output and debug logs, proposes the suggested changes for ORAS CLI commands. More examples will be appended below.
+
+### oras copy
+
+**Current debug logs** 
+
+Just pick the first two requests and responses as examples:
+
+```
+oras copy ghcr.io/oras-project/oras:v1.2.0 --to-oci-layout oras-dev:v1.2.0 --debug
+```
+
+```
+[DEBU0000] Request #0
+> Request URL: "https://ghcr.io/v2/oras-project/oras/manifests/v1.2.0"
+> Request method: "GET"
+> Request headers:
+   "User-Agent": "oras/1.2.0+Homebrew"
+   "Accept": "application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, application/vnd.oci.artifact.manifest.v1+json" 
+DEBU[0001] Response #0
+< Response Status: "401 Unauthorized"
+< Response headers:
+   "Content-Length": "73"
+   "X-Github-Request-Id": "9FC6:30019C:17C06:1C462:66AD0463"
+   "Content-Type": "application/json"
+   "Www-Authenticate": "Bearer realm=\"https://ghcr.io/token\",service=\"ghcr.io\",scope=\"repository:oras-project/oras:pull\""
+   "Date": "Fri, 02 Aug 2024 16:08:04 GMT" 
+DEBU[0001] Request #1
+> Request URL: "https://ghcr.io/token?scope=repository%3Aoras-project%2Foras%3Apull&service=ghcr.io"
+> Request method: "GET"
+> Request headers:
+   "User-Agent": "oras/1.2.0+Homebrew" 
+DEBU[0002] Response #1
+< Response Status: "200 OK"
+< Response headers:
+   "Content-Type": "application/json"
+   "Docker-Distribution-Api-Version": "registry/2.0"
+   "Date": "Fri, 02 Aug 2024 16:08:05 GMT"
+   "Content-Length": "69"
+   "X-Github-Request-Id": "9FC6:30019C:17C0D:1C46C:66AD0464" 
+```
+
+**Suggested changes:**
+
+```
+$ oras copy ghcr.io/oras-project/oras:v1.2.0 --to-oci-layout oras-dev:v1.2.0 --verbose debug
+```
+
+```
+2024-08-02 23:56:02 > msg=Request #0
+> Request URL: "https://ghcr.io/v2/oras-project/oras/manifests/v1.2.0"
+> Request method: "GET"
+> Request headers:
+   "User-Agent": "oras/1.2.0+Homebrew"
+   "Accept": "application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, application/vnd.oci.artifact.manifest.v1+json" 
+
```

Comment:
Do we want to insert separators like `-----`?

Reply:
Would recommend looking at other tools. There are so many nicer options - 
![image](https://github.com/user-attachments/assets/3a8967a2-9210-46c7-8a51-458cdbf3f4b7)

Reply:
I propose to add separator lines between each request and response for readability. This log separator format seems much more common.

Reply:
Maybe an empty line between incoming and outgoing message